src/app/pages/courses-page/courses-page.component.ts

```diff
@@ -16,6 +16,7 @@ export class CoursesPageComponent implements OnInit{
 
   ngOnInit(): void {
     this.coursesService.loading.subscribe(state => this.loading = state);
+    this.coursesService.loadAll();
```

Comment:
As I wrote above: remove the service, and here you just dispatch the action to load the courses (for example: LoadCoursesAction). Then you create some loadCourses$ effect that subscribes to this LoadCoursesAction action. Then you will implement all the logic for loading courses in this effect. Finally, when courses are loaded from the backend, the effect dispatches an action to update the courses (ex: CoursesLoaded). The reducer should update the courses for this CoursesLoaded action.
Create a variable of type Observable <Course []> and subscribe to the selectCourses selector, which returns a list of courses. Add input property courses to CoursesListComponent and expose courses with async pipe using property binding.